### PR TITLE
M2P-167 Fix error issues reported by the Magento coding standard tool in the remaining directories

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Before we had the PRs for fixing error issues in the Model, Test, view directories reported by the Magento coding standard tool
So this PR is to fix the error issues in the remaining directories

The following command to run:
vendor/bin/phpcs --standard=Magento2 app/code/Bolt/Boltpay/ -n

See the reported issues in the following link: https://gist.github.com/johnmccombs1/37053b5d5d0a3ebca44b4041230dbdcf

Fixes: https://boltpay.atlassian.net/browse/M2P-167

#changelog Fix error issues reported by the Magento coding standard tool in the remaining directories

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [x] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Helper/Cart.php [hunk 1]
```diff
@@ -61,7 +61,7 @@
 use Bolt\Boltpay\Model\ErrorResponse as BoltErrorResponse;
 use Bolt\Boltpay\Helper\MetricsClient;
 use Bolt\Boltpay\Helper\FeatureSwitch\Decider as DeciderHelper;
-
+use Magento\Framework\Serialize\Serializer\Serialize;
```
Comment (danac-gs) [inline Helper/Cart.php:64]: Not availvable before Magento 2.2. We may not officially support older Magento versions, but may still have merchant stores running on them.

@matt-thomason @njfaries @vitaliyreznikov how do we proceed here?

File: Helper/Cart.php [hunk 2]
```diff
@@ -1509,22 +1519,22 @@ public function buildCartFromQuote($quote, $immutableQuote, $items, $placeOrderP
         if ($placeOrderPayload) {
             $placeOrderPayload = json_decode($placeOrderPayload);
 
-            $billAddress          = @$placeOrderPayload->billingAddress;
-            $billingStreetAddress = (array)@$billAddress->street;
+            $billAddress          = $placeOrderPayload->billingAddress ?? null;
```
Comment (danac-gs) [inline Helper/Cart.php:1522]: `??` not backward compatible, can we use `isset` or `!empty` instead?

File: Helper/Session.php [hunk 3]
```diff
@@ -29,6 +29,7 @@
 use Magento\Framework\App\Area;
 use Magento\Framework\Data\Form\FormKey;
 use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\Serialize\Serializer\Serialize;
```
Comment (danac-gs) [inline Helper/Session.php:32]: Not backward compatible

File: Helper/Api.php [hunk 4]
```diff
@@ -256,7 +256,7 @@ public function sendRequest(Request $request)
                 $headers = $response->getHeaders();
                 $report->setMetaData([
                     'META DATA' => [
-                        'bolt_trace_id' => @$headers[ConfigHelper::BOLT_TRACE_ID_HEADER],
+                        'bolt_trace_id' => $headers[ConfigHelper::BOLT_TRACE_ID_HEADER] ?? null,
```
[the remark this answers is not in the document]
Comment (johnmccombs1) [reply Helper/Api.php:259]: @danac-gs @matt-thomason @mccottry @vitaliyreznikov @njfaries The reason I made this change is that Magento 2.1 is already on end-of-life and PHP 5.6 is pretty slow for Magento. So most of the merchants would likely upgrade to Magento 2.3 (or above) or even if they are on Magento 2.1 they would still use PHP7. I've done integration for a lot of merchants and I haven't seen any that used Magento 2.1

I can switch the syntax back to 
            $discount_code = isset($request->discount_code) ? $request->cart->discount_code : null;

But it is more verbose and I think it's now time to specify that our module is only working on PHP7 in composer.json

[the remark this answers is not in the document]
Comment (matt-thomason) [reply Helper/Api.php:259]: When I did my linting refactor I used the `isset` way, but you are right, I'd love to get off PHP5.6.
I'll bring this up against with Abhinay & Ellen

**Comment by matt-thomason** [comment]
@johnmccombs1 @danac-gs Ok, after discussing with the rest of the team, we can officially drop support for PHP5 and Magento 2.0, 2.1 and 2.2.

@johnmccombs1 You're clear to use these specific functions/methods.

**Comment by danac-gs** [comment]
@johnmccombs1 please fix conflicts

**Comment by johnmccombs1** [comment]
@danac-gs I've fixed conflicts

**Review by danac-gs — APPROVED** [review]

**Comment by johnmccombs1** [comment]
@njfaries I have fixed the conflicts. Can you re-check?

**Review by danac-gs — APPROVED** [review]

**Review by njfaries — APPROVED** [review]

**Pull request merged**